PurchasesTests/Mocks/MockSK2BeginRefundRequestHelper.swift

```diff
@@ -22,7 +22,21 @@ import StoreKit
 class MockSK2BeginRefundRequestHelper: SK2BeginRefundRequestHelper {
 
     var maybeMockSK2Error: Error?
-    var maybeMockSK2Status: StoreKit.Transaction.RefundRequestStatus?
+
+    // We can't directly store instances of StoreKit.Product, since that causes
+    // linking issues in iOS < 15, even with @available checks correctly in place.
+    // So instead, we store the underlying product as Any and wrap it with casting.
+    // https://openradar.appspot.com/radar?id=4970535809187840
+    private var untypedSK2Status: Any?
```

Comment:
we had the same issue in ProductDetails. Isn't this fun.

Reply:
oooooof. thanks for updating

Reply:
cc @NachoSoto this is what I was mentioning when it comes to the weirdness around storing SK2 objects